hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -479,6 +510,200 @@ public Hashtable<String, String> getPathStatus(final Path path,
+    while (true) {//TODO: better condition.
+      if (handleCopyInProgress(dstPath, tracingContext, copyId)) {
+        return;
+      }
+      try {
+        Thread.sleep(1000l); //Taken sleep time from AzureNativeFileSystemStore.
+      } catch (Exception e) {
+
+      }
+    }
+  }
+
+  /**
+   * Verifies if the blob copy is success or a failure or still in progress.
+   * @param dstPath path of the destination for the copying
+   * @param tracingContext object of tracingContext used for the tracing of the
+   * server calls.
+   * @param copyId id returned by server on the copy server-call. This id gets
+   * attached to blob and is returned by GetBlobProperties API on the destination.
+   *
+   * @return true if copying is success, false if it is still in progress.
+   * @throws AzureBlobFileSystemException exception returned in making server call
+   * for GetBlobProperties on the path. It can be thrown if the copyStatus is failure
+   * or is aborted.
+   */
+  @org.apache.hadoop.classification.VisibleForTesting
+  boolean handleCopyInProgress(final Path dstPath,
+      final TracingContext tracingContext,
+      final String copyId) throws AzureBlobFileSystemException {
+    BlobProperty blobProperty = getBlobProperty(dstPath,
+        tracingContext);
+    if (blobProperty != null && copyId.equals(blobProperty.getCopyId())) {
+      if (COPY_STATUS_SUCCESS.equals(blobProperty.getCopyStatus())) {
+        return true;
+      }
+      if (COPY_STATUS_FAILED.equals(blobProperty.getCopyStatus())) {
+        throw new AbfsRestOperationException(
+            COPY_BLOB_FAILED.getStatusCode(), COPY_BLOB_FAILED.getErrorCode(),
+            null,
+            new Exception(COPY_BLOB_FAILED.getErrorCode()));
+      }
+      if (COPY_STATUS_ABORTED.equals(blobProperty.getCopyStatus())) {
+        throw new AbfsRestOperationException(
+            COPY_BLOB_ABORTED.getStatusCode(), COPY_BLOB_ABORTED.getErrorCode(),
+            null,
+            new Exception(COPY_BLOB_ABORTED.getErrorCode()));
+      }
+    }
+    return false;
+  }
+
+  @org.apache.hadoop.classification.VisibleForTesting
+  String getCopyBlobProgress(final AbfsRestOperation copyOp) {
+    return getCopyStatus(copyOp.getResult());
+  }
+
+  /**
+   * Wrapper on {@link #getBlobProperty(Path, TracingContext)} with the handling
+   * for the httpStatusCode = 404 on the server response.
+   *
+   * @param blobPath path for which the property information is required
+   * @param tracingContext object of TracingContext required for the tracing of
+   * server calls
+   * @return instance of BlobProperty if the blob is present on the given path.
+   * <code>null</code> if there is no blob on the given path.
+   * @throws AzureBlobFileSystemException exception other than
+   * {@link AbfsRestOperationException} for httpStatusCode = 404 on the server
+   * response.
+   */
+  BlobProperty getBlobPropertyWithNotFoundHandling(Path blobPath,
```

Comment:
Can be handled in getBlobProperty method only, no need of separate function.

Reply:
Want getBlobProperty method to throw 404, as done by getFileStatus() (it handles not exception), The function `getBlobPropertyWithNotFoundHandling` is required where getBlobProperty method is called and want to handle 404 cases.

Reply:
has been removed in list pr. nowhere used now.